File: Code/Components/DanpheEMR.ServerModel/MasterModels/PrintExportConfigModel.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace DanpheEMR.ServerModel
{
     public class PrintExportConfigModel
    {
        [Key]
        public int PrintExportSettingsId { get; set; }
        public string SettingName { get; set; }
        public string PageHeaderText { get; set; }
        public string ReportDescription { get; set; }
        public string ModuleName { get; set; }
        public bool ShowHeader { get; set; }
        public bool ShowFooter { get; set; }
        public bool ShowUserName { get; set; }
        public bool ShowPrintExportDateTime { get; set; }
        public bool ShowNpDate { get; set; }
        public bool ShowEnDate { get; set; }
        public bool ShowFilterDateRange { get; set; }
        public bool ShowOtherFilterVariables { get; set; }
        public DateTime? CreatedOn { get; set; }
        public int? CreatedBy { get; set; }
        public DateTime? ModifiedOn { get; set; }
        public int? ModifiedBy { get; set; }
        public bool IsActive { get; set; }
    }
}

File: Code/Websites/DanpheEMR/wwwroot/DanpheApp/src/app/settings-new/shared/print-export-config.model.ts
import { FormBuilder, FormGroup, Validators } from "@angular/forms";

export class PrintExportConfigurationModel {

  public PrintExportSettingsId: number = 0;
  public SettingName: string = null;
  public PageHeaderText: string = null;
  public ReportDescription: string = null;
  public ModuleName : string = null;
  public ShowHeader:boolean = true;
  public ShowFooter : boolean = true;
  public ShowUserName : boolean = true;
  public ShowPrintExportDateTime :boolean = true;
  public ShowNpDate : boolean = true;
  public ShowEnDate : boolean = true;
  public ShowFilterDateRange : boolean = true;
  public ShowOtherFilterVariables :boolean = true;
  public IsActive: boolean = true;
  public CreatedBy: number = null;
  public ModifiedBy: number = null;
  public CreatedOn: string = null;
  public ModifiedOn: string = null;
  public ConfigurationValidator: FormGroup = null;


  constructor() {
    var _formBuilder = new FormBuilder();
    this.ConfigurationValidator = _formBuilder.group({
       'SettingName': ['', Validators.compose([Validators.required])],
       'PageHeaderText': ['', Validators.compose([Validators.required])],
       'ModuleName': ['', Validators.compose([Validators.required])],
      //  'ReportDescription': ['', Validators.compose([Validators.required])],
    });
  }

  public IsDirty(fieldName): boolean {
    if (fieldName == undefined) {
      return this.ConfigurationValidator.dirty;
    }    
    else
      return this.ConfigurationValidator.controls[fieldName].dirty;
  }

  public IsValid(): boolean {
    if (this.ConfigurationValidator.valid) {
      return true;
    }
    else {
      return false;
    }
  }

  public IsValidCheck(fieldName, validator): boolean {
    if (fieldName == undefined) {
      return this.ConfigurationValidator.valid;
    }
    else {
      return !(this.ConfigurationValidator.hasError(validator, fieldName));
    }  
  }
}
